text/0757-default-modifier-manager.md

```diff
@@ -0,0 +1,276 @@
+---
+Stage: Accepted
+Start Date: 2021-05-17
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): Ember.js, Learning
+RFC PR: https://github.com/emberjs/rfcs/pull/757
+---
+
+# Default Modifier Manager
+
+## Summary
+
+Anything that can be in a template has its lifecycle managed by the manager pattern.
+Today, we have known managers for `@glimmer/component`, `@ember/helper`, etc.
+But what happens when the VM encounters an object for which there is no manager?,
+such as a plain function? This RFC explores and proposes a default behavior for
+those unknown scenarios when it comes to _modifiers_.
```

Comment:
I'd suggest summarizing what we're actually doing here: this should be a TL;DR for the rest of it.

That might look something like this:

---

Enable standalone functions to be used a modifiers, by introducing a default *modifier manager*.

Today, when defining a modifier to set scroll position when a condition is true—

```hbs
<div {{scroll when=@someCondition}}></div>
```

—we need to import `modifier` and define it this way:

```js
// app/modifiers/scroll.js
import { modifier } from 'ember-modifier';

export default modifier(function scroll(el, _, { when: shouldScroll }) {
  if (shouldScroll) {
    el.scrollTop(0);
  }
});
```

With the default modifier manager proposed in this RFC, we can simply write a function which receives the element and the arguments directly:

```js
// app/modifiers/scroll.js
export default function scroll(el, { when: shouldScroll }) {
  if (shouldScroll) {
    el.scrollTop(0);
  }
}
```

This complements the introduction of a [default helper manager][rfc-0756], and is especially nice when combined with strict mode templates via [ember-template-imports][eti] (or as proposed in [First Class Component Templates][fcct]), where functions.

[rfc-0756]: https://github.com/emberjs/rfcs/pull/756
[eti]: https://github.com/ember-template-imports/ember-template-imports
[fcct]: https://github.com/emberjs/rfcs/pull/779

---

This way you tell people what they *get* from this!

Reply:
added!